src/components/img-with-placeholder/index.js

```diff
@@ -139,32 +151,40 @@ export default class Img extends React.PureComponent {
   }
 
   handleImageLoaded() {
-    if (!this.state.isLoaded) {
+    // Progressive image
+    // Let user see the blur image,
+    // and slowly make the blur image clearer
+
+    // in order to make sure users see the blur image,
+    // delay the clear image rendering
+    setTimeout(() => {
       this.setState({
         isLoaded: true,
       })
-      setTimeout(() => {
-        if (this._isMounted) {
-          this.setState({
-            toShowPlaceholder: false,
-          })
-        }
-      }, 1000)
-    }
+    }, 500)
+
+    // after clear image rendered, not display placeholder anymore
+    setTimeout(() => {
+      this.setState({
+        toShowPlaceholder: false,
+      })
+    }, 1500)
```

Comment:
Please help to fix the problem that this component may invoke `setState` after unmounted.

```js
if (this._mounted) {
  this.setState({
  // ...
  }))
}
```